### PR TITLE
[Compute] Bug fix for Update-Azvmss

### PR DESCRIPTION
**Pull request by haagha**
## Description

[#22195]
## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Compute/Compute/ChangeLog.md [hunk 1]
```diff
@@ -23,6 +23,7 @@
 * Added `-Hibernate` switch parameter to `Stop-AzVmss` default parameter set. 
 * For `Get-AzVmRunCommand`, a bug is fixed to work when returning a list of RunCommands [#22403]
 * Updated Azure.Core to 1.34.0.
+* Fix for `Update-AzVMSS` to update ImageReferenceSKU [#22195]
```
Comment (YanaXu) [inline src/Compute/Compute/ChangeLog.md:26]: Update "Fix" to "Fixed".

**Comment by YanaXu** [comment]
Hi @haagha ， please fix failed tests and update the changelog.

File: src/Compute/Compute/ChangeLog.md (hunk 1 above)
Comment (haagha) [reply src/Compute/Compute/ChangeLog.md:27]: Done

Comment (YanaXu) [reply src/Compute/Compute/ChangeLog.md:27]: Hi @haagha , please double-check the changes. You didn't merge it correctly.

**Comment by haagha** [comment]
Hi @YanaXu added the changes. waiting for CI checks to complete

**Comment by YanaXu** [comment]
Hi @haagha, could you fix failed tests and resolve the conflict in ChangeLog?
And FYI, the code complete date for this sprint is today. If this PR is not merged today, we'll release it in the next sprint.

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetUpdateMethod.cs [hunk 2]
```diff
@@ -64,7 +64,8 @@ public override void ExecuteCmdlet()
                     if (this.VirtualMachineScaleSet != null
                             && this.VirtualMachineScaleSet.VirtualMachineProfile != null
                             && this.VirtualMachineScaleSet.VirtualMachineProfile.StorageProfile != null
-                            && this.VirtualMachineScaleSet.VirtualMachineProfile.StorageProfile.ImageReference != null)
+                            && this.VirtualMachineScaleSet.VirtualMachineProfile.StorageProfile.ImageReference != null
+                            && this.VirtualMachineScaleSet.VirtualMachineProfile.StorageProfile.ImageReference.Id != null)
```
Comment (Sandido) [inline src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetUpdateMethod.cs:68]: Could you describe the issue that caused this bug and how this edit fixes it?

**Review by Sandido — CHANGES_REQUESTED** [review]

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1 [hunk 3]
```diff
@@ -450,7 +450,7 @@ function Test-VirtualMachineScaleSetUpdate
         $adminUsername = 'Foo12';
         $adminPassword = $PLACEHOLDER;
 
-        $imgRef = Get-DefaultCRPImage -loc $loc;
```
Comment (Sandido) [inline src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1:453]: where is the test for the specific bug this edit is fixing?

**Review by Sandido — CHANGES_REQUESTED** [review]

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetUpdateMethod.cs (hunk 2 above)
Comment (haagha) [reply src/Compute/Compute/Generated/VirtualMachineScaleSet/VirtualMachineScaleSetUpdateMethod.cs:68]: The issue is here @adam https://github.com/Azure/azure-powershell/issues/22195

The IF condition nulls the imagereference even when imagereference is being updated with normal rbac based ARM image using Imagereference.ID

It's pretty straightforward, imagereference.Id is passed for a new image, but the old IF condition did not check for that property and went into the Condition nullifiying imagereferece. Hence User's SKU was not updated.

I've already tested the fix. Let me know if there is further explanation needed.

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1 (hunk 3 above)
Comment (haagha) [reply src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1:453]: test has been added, there was no existing test for this.

**Comment by haagha** [comment]
@YanaXu  I would like to merge in this release, since this is an impactful bug. Added test and fixed failures. Running CI checks now.

**Comment by YanaXu** [comment]
Hi @haagha ,the code complete date was yesterday. This PR is still not ready. Please double-check changelog.  There are errors in it.  We can't include it in this sprint. Please hold it to the next sprint.

File: src/Compute/Compute/ChangeLog.md [hunk 4]
```diff
@@ -25,7 +25,11 @@
 * Updated Azure.Core to 1.34.0.
 * Added new cmdlets `Get-AzHostSize` and `Update-AzHost`.
 * Added the `Standard` value to the `SecurityType` parameter to the cmdlets `Set-AzDiskSecurityType`, `New-AzvmssConfig`, `Set-AzVmssSecurityProfile`, `Update-AzVmss`, `New-AzVmss`, `New-AzVMConfig`, `Set-AzVMsecurityProfile`, and `New-AzVM`.
+<<<<<<< HEAD
+* Fixed `Update-AzVMSS` to update ImageReferenceSKU [#22195]
+=======
 * Updated the above change to include `New-AzVMConfig` as 1 scenario was initially missed when only using this cmdlet.
+>>>>>>> origin/main
```
Comment (YanaXu) [inline src/Compute/Compute/ChangeLog.md:32]: Please handle this merge conflict.

Comment (haagha) [reply src/Compute/Compute/ChangeLog.md:32]: ok

Comment (haagha) [reply src/Compute/Compute/ChangeLog.md]: pushed the commit

**Comment by YanaXu** [comment]
Hi @haagha , I can't merge this PR since @Sandido didn't approve it. I have to leave this PR to the next sprint.

**Pull request merged**